src/main/java/edu/cornell/kfs/module/ar/document/web/struts/CuContractsGrantsInvoiceDocumentAction.java

```diff
@@ -146,6 +147,23 @@ protected void validateBillingPeriodDateRange(ContractsGrantsInvoiceDocument con
             warningMessages.add(configurationService.getPropertyValueAsString(CUKFSKeyConstants.WARNING_CINV_BILLING_PERIOD_START_DATE_BEFORE_AWARD_START_DATE));
         }
     }
+    
+    private boolean shouldValidateBillingPeriodDateRange(ContractsGrantsInvoiceDocument contractsGrantsInvoiceDocument) {
+        String frequencyCode = contractsGrantsInvoiceDocument.getInvoiceGeneralDetail().getAward().getBillingFrequency().getFrequency(); 
```

Comment:
Are any null checks or refreshReferenceObject() calls needed on any of the objects in this method chain?

Reply:
good point, I will investigate

Reply:
above in the function call, the invoice general detail is referenced from the document.  The getAward function makes a call to Spring to get the award.  So  I think a null check on the award is appropriate.  I also found the frequency code as a member variable on the award, so I don't need to get the frequency object first.